### ISSUE TITLE
Deleted thread continues to show replies button

### ISSUE THREAD

**Issue by amottier** [RocketChat/Rocket.Chat#21560]
### Description:

When a message that was used to initiate a thread is deleted message is lost but the number of replies and the reply button continues to be visible.

This issue is actually same as #16320.

### Steps to reproduce:

1. Write a message
1. Reply to the message to start a thread
1. Delete the message first message
1. Click on the reply button that will let you post a reply to "nothing"

### Expected behavior:

Number of replies and Reply button should not be visible anymore.

### Actual behavior:

Number of replies (even if all were deleted) is still visible and Reply button is clickable.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.12.3
- Operating System: Debian 10 Buster
- Deployment Method: Docker
- Number of Running Instances: 1
- NodeJS Version: 12.18.4
- MongoDB Version: 4.0.16

### Client Setup Information

- Browser Version: Firefox 78.9.0esr
- Operating System: Debian 10 Buster

**Comment by rahljoshi** [comment]
So when we delete the message, it should delete the whole thread, right? and it should not show a reply button because our message is deleted, which means it should simply show the message deleted.

**Comment by yash-rajpal** [comment]
@amottier I think this is the done on purpose, if the main message is deleted, the replies to it still should be visible for others to read. We should not delete whole thread when deleting the thread initiating message.

Also #16320 is a different issue where the thread messages are not visible after deleting the thread initiating message, I think the current flow solves that issue.

What do you think should be the correct flow?

**Comment by amottier** [comment]
@yash-rajpal I think I would expect the behavior described by @rahljoshi : deleting a message that initiate a thread should delete all messages created as replies to this initial thread message. And as the message that initiate a thread no longer exist I don't see the point of being able to keep the discussion going by replying to it.

But I understand this might not be the expected behavior for everyone. Such logic can give too much power to the user who initiate the thread because, by deleting his message, he will be able to delete replies that it does not authored.

**Comment by yash-rajpal** [comment]
@amottier Yes, correct, it can depend from user to user, maybe we can have a setting for this in admin panel, if replies for deleted message has to be shown or not.
Maybe someone from team can tell us more about this. @ggazzo @sampaiodiego :)

**Comment by sampaiodiego** [comment]
thanks guys.. and you right @yash-rajpal , this is the designed behavior, since a user can delete his own messages, but cannot delete other user's messages..

I'm not sure if adding a setting to have multiple behaviors would help or make things even more confusing, maybe we could explain what is going to happen showing a confirmation to the user? this way no expectation would be hurt.. let me tag @milton-rucks  to take a look on this..

**Comment by rahljoshi** [comment]
Or I think we should add functionality whenever we want to delete there should be 2 options one be like delete the whole thread and the other one be like delete only your message, I think it can work this way?

**Comment by sennewood** [comment]
Not deleting the whole thread is not a bug?!

We often have to deal with this problem.

**Comment by sennewood** [comment]
> Or I think we should add functionality whenever we want to delete there should be 2 options one be like delete the whole thread and the other one be like delete only your message, I think it can work this way?

This would be nice.
